scripts/gprdemomarglik.py

```diff
@@ -0,0 +1,140 @@
+import pyprobml_utils as pml
+#!pip install GPy
+import numpy as np
+import matplotlib.pyplot as plt 
+import GPy
+from numpy.linalg import cholesky, det, inv, lstsq
+from scipy.linalg import solve_triangular
+import scipy.spatial.distance as spdist
+from scipy import linalg
+from scipy.optimize import minimize
+
+# Dataset according to matlab code
+n = 7
+np.random.seed(20) 
+xs = 15*(np.random.uniform(low=0, high=1, size=n) - 0.5).reshape((-1, 1))
+sigma_y = 0.1
+def K(p, q):
+  p = p.transpose()
+  p = p.transpose()
+  p = np.tile(p, len(q))
+  q = np.tile(q, len(p))
+  r = 0.5*np.square(p-q)
+  return np.exp(r)
+w = K(xs, xs) + (sigma_y**2)*np.eye(n)
+w = w.conjugate()
+fs = np.linalg.cholesky(w).dot(np.random.randn(n, 1)).reshape((-1, 1))
+
+#xs = np.array([-1.3089, 6.7612, 1.0553, -1.1734, -2.9339, 7.2530, -6.5843]).reshape(-1, 1)
+#fs = np.array([1.6218, 1.8558, 0.4102, 1.2526, -0.0133, 1.6380, 0.2189]).reshape(-1, 1)
+
+n=41
```

Comment:
@Drishttii, have you tried playing with this parameter? I tried to run your code in Colab, but it got stuck for some reason. 😕

Reply:
@gerdm, I will try that and see what happens .... also let me know what error you are getting .... whenever you have time :) Thank you for the suggestion.

Reply:
For sure! I will try again and let you know.